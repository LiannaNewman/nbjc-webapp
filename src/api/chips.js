export default [{
  name: 'Accessible',
  value: '1',
}, {
  name: 'ATM Available',
  value: '2',
}, {
  name: 'ASL Spoken',
  value: '3',
}, {
  name: 'Black Friendly',
  value: '4',
}, {
  name: 'Black Owned',
  value: '5',
}, {
  name: 'Gender-Neutral Restroom',
  value: '6',
}, {
  name: 'Inclusive',
  value: '7',
}, {
  name: 'LGBTQIA+ Friendly',
  value: '8',
}, {
  name: 'Multilingual',
  value: '9',
}, {
  name: 'Person of Color owned',
  value: '10',
}, {
  name: 'Price Conscious',
  value: '11',
}, {
  name: 'Trans-Trained',
  value: '12',
}, {
  name: 'Wheelchair Ramp',
  value: '13',
}];
