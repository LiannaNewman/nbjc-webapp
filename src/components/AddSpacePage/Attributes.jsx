import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import useMediaQuery from '@material-ui/core/useMediaQuery';

import { Button, Typography } from '@material-ui/core';

import ChipFilters from '../ChipFilters';
import { chipType, addSpaceProps as addSpacePropTypes } from '../../types';

const styles = (theme) => ({
  chipWrapper: {
    display: 'block',
  },
  chip: {
    [theme.breakpoints.up('xs')]: {
      margin: '10px 10px 0 0',
    },
    [theme.breakpoints.up('mobile')]: {
      margin: '30px 40px 0 0',
    },
  },
  footer: {
    margin: '40px 0',
  },
  submitButton: {
    float: 'right',
    marginBottom: 20,
  },
});

const Attributes = ({
  chips,
  onBack,
  onNext,
  classes,
  addSpaceProps,
}) => {
  const matches = useMediaQuery('(min-width:376px)');
  const { chips: initialChips = [] } = addSpaceProps;
  const initialChipMap = initialChips.reduce((acc, c) => {
    acc[c.value] = c;
    return acc;
  }, {});
  const [activeChips, setActiveChips] = useState(
    chips.map((c) => ({ ...c, isSelected: !!initialChipMap[c.value] })),
  );
  const onChipSelected = (i) => {
    const oldChips = [...activeChips];
    oldChips[i].isSelected = !oldChips[i].isSelected;
    setActiveChips(oldChips);
  };

  const isAnySelected = () => ((activeChips.filter((c) => c.isSelected) || []).length > 0);

  const onSubmit = () => {
    const selectedChips = activeChips.filter((chip) => chip.isSelected);
    onNext({ chips: selectedChips });
  };

  return (
    <>
      <Typography variant="subtitle1" align="center">
        What are the attributes that make this an OurGuide space?
      </Typography>
      <Typography variant="h6">Select all that apply</Typography>
      <ChipFilters
        chips={activeChips}
        onChipSelected={onChipSelected}
        chipSize="medium"
        overrideClasses={{ wrapper: classes.chipWrapper, chip: classes.chip }}
      />
      <div className={classes.footer}>
        <Button
          type="submit"
          variant="contained"
          color="secondary"
          className={classes.submitButton}
          fullWidth={!matches}
          onClick={onSubmit}
          disabled={!isAnySelected()}
          disableElevation
        >
          Next
        </Button>
        <Button
          type="cancel"
          variant="outlined"
          color="secondary"
          fullWidth={!matches}
          onClick={onBack}
          disableElevation
        >
          Back
        </Button>
      </div>
    </>
  );
};

Attributes.propTypes = {
  chips: PropTypes.arrayOf(chipType).isRequired,
  onBack: PropTypes.func.isRequired,
  onNext: PropTypes.func.isRequired,
  addSpaceProps: PropTypes.shape(addSpacePropTypes),
};

Attributes.defaultProps = {
  addSpaceProps: {},
};

export default withStyles(styles)(Attributes);
