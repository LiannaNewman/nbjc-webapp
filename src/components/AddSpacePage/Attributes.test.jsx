import React from 'react';
import { render } from '@testing-library/react';

import chips from '../../api/chips';

import Attributes from './Attributes';

describe('Attributes', () => {
  test('renders with default values', () => {
    const { asFragment } = render(
      <Attributes onNext={jest.fn()} onBack={jest.fn()} chips={chips} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
