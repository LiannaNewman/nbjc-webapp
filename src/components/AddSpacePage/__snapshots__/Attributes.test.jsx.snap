// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attributes renders with default values 1`] = `
<DocumentFragment>
  <h6
    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter"
  >
    What are the attributes that make this an OurGuide space?
  </h6>
  <h6
    class="MuiTypography-root MuiTypography-h6"
  >
    Select all that apply
  </h6>
  <div
    class="ChipFilters-wrapper-6 Attributes-chipWrapper-1"
  >
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Accessible
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          ATM Available
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          ASL Spoken
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Black Friendly
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Black Owned
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Gender-Neutral Restroom
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Inclusive
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          LGBTQIA+ Friendly
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Multilingual
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Person of Color owned
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Price Conscious
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Trans-Trained
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiButtonBase-root MuiChip-root ChipFilters-chip-5 Attributes-chip-2 MuiChip-colorPrimary MuiChip-clickableColorPrimary MuiChip-outlined MuiChip-outlinedPrimary MuiChip-clickable"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiChip-label"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Wheelchair Ramp
        </p>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
  <div
    class="Attributes-footer-3"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained Attributes-submitButton-4 MuiButton-containedSecondary MuiButton-disableElevation Mui-disabled MuiButton-fullWidth Mui-disabled"
      disabled=""
      tabindex="-1"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        Next
      </span>
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-disableElevation MuiButton-fullWidth"
      tabindex="0"
      type="cancel"
    >
      <span
        class="MuiButton-label"
      >
        Back
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
